Lib/asyncio/taskgroup.py

```
@@ -0,0 +1,282 @@
+#
+# This source file is part of the EdgeDB open source project.
```

Hmm... @1st1, does all this need to stay in the file in case we copy it into the stdlib?

---

No, I'll contribute it in a clean way under PSFL.

---

Also, FWIW, the current code still doesn't work with EGs properly. If an EG contains a CancelledError, the TaskGroup would propagate the CancelledError, not the EG (thus discarding all other errors).  To fix this we'll need to fix the asyncio itself to recognize CancelledErrors wrapped in a EG. I'll do that as soon as we have some basic high-level API.

---

Which makes me think that taskgroup.py doesn’t belong in this PR.

---

You're right, I copied it over from the previous branch but will remove.